source/tests/system/nidaqmx_driver_api_tests.cpp

```diff
@@ -1956,6 +1957,41 @@ TEST_F(NiDAQmxDriverApiTests, ReadAnalogWaveforms_WithTimingAndExtendedPropertie
   }
 }
 
+TEST_F(NiDAQmxDriverApiTests, ReadAnalogWaveforms_ReadAllAvailable_ReturnsAvailableSamples)
+{
+  const auto TIMEOUT = 10.0;
+  CreateAIVoltageChanResponse create_channel_response;
+  auto create_channel_status = create_ai_voltage_chan("gRPCSystemTestDAQ/ai0", "ai0", -1.0, 1.0, create_channel_response);
+  EXPECT_SUCCESS(create_channel_status, create_channel_response);
+  create_channel_status = create_ai_voltage_chan("gRPCSystemTestDAQ/ai1", "ai1", -1.0, 1.0, create_channel_response);
+  EXPECT_SUCCESS(create_channel_status, create_channel_response);
+
+  start_task();
+  ::grpc::ClientContext context;
+  ReadAnalogWaveformsResponse read_response;
+  auto read_status = read_analog_waveforms(context, READ_ALL_AVAILABLE, TIMEOUT, WaveformAttributeMode::WAVEFORM_ATTRIBUTE_MODE_NONE, read_response);
+  stop_task();
+
+  EXPECT_SUCCESS(read_status, read_response);
+  EXPECT_EQ(read_response.status(), DAQMX_SUCCESS);
+  EXPECT_EQ(read_response.waveforms_size(), 2);
+  EXPECT_EQ(read_response.samps_per_chan_read(), 1);
```

Comment:
Configure finite sample clock timing and validate that the sample count matches the number you passed to the timing function.

This also applies to the digital test.